src/SFA.DAS.EAS.Employer_Financial.Database/StoredProcedures/GetAllTransactionDetailsForAccountByDate.sql

```diff
@@ -56,6 +56,59 @@ AND DateCreated >= @FromDate
 AND DateCreated < @ToDate
 
 
+
+-- We will use this View to get payments that relate to transaction lines and are funding source 1,2,3
+;WITH AllFunds (DateCreated, AccountId, Ukprn, Uln, PeriodEnd, PaymentId, FundingSource, Amount, PaymentMetaDataId, TransactionTypeDesc,
+	TrainingProvider,
+	Apprentice,
+	ApprenticeTrainingCourse,
+	ApprenticeTrainingCourseLevel)
+AS
+(
+--	SELECT PaymentId, FundingSource, Amount, p.AccountId, p.Ukprn, p.Uln, p.PeriodEnd, p.PaymentMetaDataId
+	SELECT 
+			transLine.DateCreated,
+			p.AccountId, 
+			p.Ukprn, 
+			p.Uln, 
+			p.PeriodEnd, 
+			PaymentId, 
+			FundingSource, 
+			Amount, 
+			p.PaymentMetaDataId, 
+			ptt.[Description] AS TransactionTypeDesc,
+			meta.ProviderName AS TrainingProvider,
+			meta.ApprenticeName AS Apprentice,
+			meta.ApprenticeshipCourseName AS ApprenticeTrainingCourse,
+			meta.ApprenticeshipCourseLevel AS ApprenticeTrainingCourseLevel
+	from [employer_financial].[Payment]   p
+		INNER JOIN (SELECT PeriodEnd,AccountId,ukprn, EmpRef, TransactionDate, DateCreated, LevyDeclared, EnglishFraction 
```

Comment:
It isn't clear why a correlated sub query is being used here rather than a normal join with the date filtering in the sub-query in the where clause. The query would be simpler - the where conditions would need qualifying with the transline alias

Reply:
I have simplified this